Add a filter to remove mutation

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #52614 Avoid dispatch overhead in call to mkldnn convolution
* #52613 Include max pool in fusion groups
* #52513 Extend subgraph utils to cover merging a node following a subgraph
* #52512 Remove unused subgraph vmap api
* #51924 Only functionalize ops which we want to include in mkldnn group
* **#51923 Add a filter to remove mutation**
* #51922 Dont error on 0-dim in convolution
* #51921 Add 0-dim support for binary mkldnn ops
* #51736 Handle mkldnn broadcasting in mkldnn fuser
* #51600 Add MKLDNN fuser
* #52786 [NNC] Make NNC sanitize input names
* #51484 Add support for linear in mkldnn fusion
* #51483 Move conv to mkldnn

Differential Revision: [D26696700](https://our.internmc.facebook.com/intern/diff/D26696700)

Commit:
Add a filter to remove mutation

[ghstack-poisoned]

Commit:
Update on "Add a filter to remove mutation"




[ghstack-poisoned]

ZolotukhinM commented on test/cpp/jit/test_misc.cpp:
```diff
@@ -2376,5 +2377,25 @@ TEST(ComputeFlopsTest, Basic) {
+  %9 : int = prim::Constant[value=4]() # /data/users/eellison/pytorch/test/jit/test_remove_mutation.py:79:20
+  %x.3 : Tensor = aten::add(%x.1, %2, %2) # /data/users/eellison/pytorch/test/jit/test_remove_mutation.py:76:16
+  %7 : Tensor = aten::add_(%x.3, %2, %2) # /data/users/eellison/pytorch/test/jit/test_remove_mutation.py:77:12
+  %y.1 : Tensor = aten::add(%x.3, %9, %2) # /data/users/eellison/pytorch/test/jit/test_remove_mutation.py:79:16
```
Nit: remove the comments to make it easier to read :)

ZolotukhinM reviewed (approved):
Looks good to me!

Krovatkin reviewed (approved):
:shipit:

Commit:
Update on "Add a filter to remove mutation"




[ghstack-poisoned]

Commit:
Update on "Add a filter to remove mutation"




[ghstack-poisoned]

Commit:
Update on "Add a filter to remove mutation"




[ghstack-poisoned]

Commit:
Update on "Add a filter to remove mutation"




[ghstack-poisoned]

Commit:
Update on "Add a filter to remove mutation"




[ghstack-poisoned]

Krovatkin reviewed (approved):
:shipit:

Commit:
Update on "Add a filter to remove mutation"




[ghstack-poisoned]

Commit:
Update on "Add a filter to remove mutation"




[ghstack-poisoned]

Commit:
Update on "Add a filter to remove mutation"




[ghstack-poisoned]

Commit:
Update on "Add a filter to remove mutation"




[ghstack-poisoned]

Commit:
Update on "Add a filter to remove mutation"


Differential Revision: [D26696700](https://our.internmc.facebook.com/intern/diff/D26696700)

[ghstack-poisoned]
